### PR TITLE
Feature NEX 1243 - Add support custom error

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to**

https://oat-sa.atlassian.net/browse/NEX-1243

**Related to this PR**

https://github.com/oat-sa/tao-core/pull/2580

**Description**

Support specific error `999` for handle in FE without errors

### CODE REVIEW

File: views/js/controller/backoffice.js [hunk 1]
```diff
@@ -33,6 +33,25 @@ define([
 ], function ($, _, __, context, helpers, router, uikitLoader, history, feedback, logoutEvent) {
     'use strict';
 
+    function checkAjaxResponse(ajaxResponse) {
+        if (ajaxResponse && ajaxResponse !== null) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (andrey-ponamarev) [inline views/js/controller/backoffice.js:41]:
```suggestion
        return ajaxResponse && ajaxResponse !== null;
```

File: views/js/controller/backoffice.js [hunk 2]
```diff
@@ -33,6 +33,25 @@ define([
...
+
+    function checkAjaxResponseProperties(ajaxResponse) {
+        if (typeof ajaxResponse.success !== 'undefined' &&
+            typeof ajaxResponse.type !== 'undefined' &&
+            typeof ajaxResponse.message !== 'undefined' &&
+            typeof ajaxResponse.data !== 'undefined') {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (andrey-ponamarev) [inline views/js/controller/backoffice.js:52]:
```suggestion
        return typeof ajaxResponse.success !== 'undefined' &&
            typeof ajaxResponse.type !== 'undefined' &&
            typeof ajaxResponse.message !== 'undefined' &&
            typeof ajaxResponse.data !== 'undefined'
```

File: views/js/controller/backoffice.js [hunk 3]
```diff
@@ -73,29 +92,26 @@ define([
...
-                            typeof ajaxResponse.message !== 'undefined' &&
-                            typeof ajaxResponse.data !== 'undefined'
-                        ) {
-                            errorMessage = `${request.status}: ${ajaxResponse.message}`;
-                        } else {
-                            errorMessage = `${request.status}: ${request.responseText}`;
-                        }
-                    } catch (err) {
-                        // It does not seem to be valid JSON.
+                    if (checkAjaxResponse() && checkAjaxResponseProperties()) {
```
Comment (andrey-ponamarev) [inline views/js/controller/backoffice.js:112]: Could you rename function to `hasAjaxResponse` and `hasAjaxResponseProperties ` ?

Comment (habemuscode) [reply views/js/controller/backoffice.js]: Done

**Pull request merged**